src/server/debug/server/log.go

```diff
@@ -0,0 +1,188 @@
+			log.Info(ctx, "reverted grpc log level", zap.String("from", from), zap.String("to", to))
+		}
+		switch x.Grpc {
+		case debug.SetLogLevelRequest_DEBUG:
+			log.SetGRPCLogLevelFor(zap.DebugLevel, d, notifyRevert)
+			log.Info(ctx, "set grpc log level to debug", zap.Duration("revert_after", d))
+		case debug.SetLogLevelRequest_INFO:
+			log.SetGRPCLogLevelFor(zap.InfoLevel, d, notifyRevert)
+			log.Info(ctx, "set grpc log level to info", zap.Duration("revert_after", d))
+		case debug.SetLogLevelRequest_ERROR:
+			log.SetGRPCLogLevelFor(zap.ErrorLevel, d, notifyRevert)
+			log.Info(ctx, "set grpc log level to error", zap.Duration("revert_after", d))
+		case debug.SetLogLevelRequest_OFF:
+			log.SetGRPCLogLevelFor(zap.FatalLevel, d, notifyRevert)
+			log.Info(ctx, "set grpc log level to fatal", zap.Duration("revert_after", d))
+		default:
+			return result, status.Errorf(codes.InvalidArgument, "cannot set grpc log level to %v", x.Grpc.String())
+		}
+	case *debug.SetLogLevelRequest_Pachyderm:
+		switch x.Pachyderm {
+		case debug.SetLogLevelRequest_DEBUG:
+			log.SetLevelFor(log.DebugLevel, d, func(from, to string) {
+				log.Info(ctx, "reverted log level", zap.String("from", from), zap.String("to", to))
+			})
+			log.Info(ctx, "set log level to debug", zap.Duration("revert_after", d))
+		case debug.SetLogLevelRequest_INFO:
+			log.SetLevelFor(log.InfoLevel, d, func(from, to string) {
+				log.Info(ctx, "reverted log level", zap.String("from", from), zap.String("to", to))
+			})
+			log.Info(ctx, "set log level to info", zap.Duration("revert_after", d))
+		case debug.SetLogLevelRequest_ERROR:
+			log.SetLevelFor(log.ErrorLevel, d, func(from, to string) {
+				log.Error(ctx, "reverted log level", zap.String("from", from), zap.String("to", to))
+			})
+			log.Error(ctx, "set log level to error", zap.Duration("revert_after", d))
+		default:
+			return result, status.Errorf(codes.InvalidArgument, "cannot set log level to %v", x.Pachyderm.String())
+		}
+	}
+
+	if s.sidecarClient == nil {
+		result.AffectedPods = append(result.AffectedPods, s.name)
+	} else {
+		result.AffectedPods = append(result.AffectedPods, s.name+".user")
+	}
+
+	// If this is the worker server, also adjust the storage sidecar.
+	if cc := s.sidecarClient; cc != nil {
+		tctx, c := context.WithTimeout(ctx, 5*time.Second)
+		if _, err := cc.DebugClient.SetLogLevel(propagateMetadata(tctx), req); err != nil {
+			result.ErroredPods = append(result.ErroredPods, fmt.Sprintf("%s.storage(%v)", s.name, err))
+		} else {
+			result.AffectedPods = append(result.AffectedPods, s.name+".storage")
+		}
+		c()
+		return result, nil
+	}
+
+	if !req.GetRecurse() {
+		// If not recursive mode, return now.
+		return result, nil
+	}
+	req.Recurse = false
+
+	// Recurse to other pachyderm processes.
+	pods := map[string]string{}
+	apps := map[string]string{
+		"pach-enterprise": strconv.Itoa(int(s.env.Config().Port)),
+		//"pachw": strconv.Itoa(int(s.env.Config().PeerPort)),
```

Comment:
Commented?

Reply:
I will delete.  I really want pachw to participate in this, but we can't authenticate our calls because pachw doesn't have an enterprise server to check validation status.  (Though honestly, pachw needs more thought.  If pachd spins it up while in extra-debugging mode, should pachw produce extra debug logs?  Probably yes.  But then we'd need extra startup environment variables; base log level, enhanced log level, remaining duration on the enhanced log level.  Well, to really get it right we'd need that.  I suppose workers suffer from the same problem, too.  Will need a mini-design.)